### ISSUE TITLE
Link should not prevent click events when no HREF

### ISSUE THREAD

**Issue by dabrad26** [carbon-design-system/carbon#6251]
When the <Link> component has no href a CSS method of disabling click is used.  This should probably be done via the JS if anything and not done via CSS.  It seems if a <Link has href respect it if not and has onClick use it.  And finally disable it if nothing (the click event shouldn't even be needed since the default style anchor with no href should be used.

```
.bx--link:not([href]):not(button) {
color: #c6c6c6;
cursor: not-allowed;
pointer-events: none;
-ms-touch-action: none;
touch-action: none;
}
```

Expected Behavior:

Remove the pointer-events and touch-action and if onClick is passed in set cursor:pointer on the Link.

There might be some back story to this I don't know.  So would like to find out why this is done.  There are several design requests to use link style for internal navigation actions. And also for those using Segment or tracking on click events or something like HTML5 history in react we want to call that verse calling an actual browser href.

Team: IBM Aspera

**Comment by dabrad26** [comment]
@superkhau

**Comment by superkhau** [comment]
+1, we ran into this issue on API Connect -- Had to work around it with a suggestion from @dabrad26 by adding an `href` prop (even though we're not linking anywhere, just triggering an `onClick` event).

To be more specific, if you don't add `href`, you get an unclickable link that looks like:

![image](https://user-images.githubusercontent.com/1617364/84323425-a3e72780-ab2b-11ea-8bdb-c62f2647d6cf.png)

# TLDR
We should mark `href` as required to get `onClick` to work properly OR remove the requirement of having to define `href` at all. Also, we should at least document the reasoning behind the current behaviour at https://www.carbondesignsystem.com/components/link/usage for posterity.
